microcosm_pubsub/published_message.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from time import time
+from typing import Dict
+
+from microcosm.api import binding
+
+from microcosm_pubsub.constants import PUBLISHED_KEY
+
+
+@dataclass
+class PublishedMessage:
+    """
+    Container class encapsulating all of the necessary components for
+    publishing a given message
+
+    """
+    media_type: str
+    message: str
+    opaque_data: dict
+
+    # Deprecated: those attributes are SNS-specific and will be removed from this class
+    message_attributes: Dict[str, Dict[str, str]] = None
+    topic_arn: str = None
+
+
+@binding("published_message_builder")
+class PublishedMessageBuilder:
+    def __init__(self, graph):
+        self.opaque = graph.opaque
+        self.pubsub_message_schema_registry = graph.pubsub_message_schema_registry
+
+    def __call__(self, media_type, uri=None, opaque_data=None, **kwargs) -> PublishedMessage:
```

Comment:
This code is basically a copy of `sns_producer.create_message`, after the pieces about `topic_arns` and `message_attributes` have been removed.